setroot.c

```diff
@@ -955,6 +995,26 @@ parse_opts( unsigned int argc, char **args )
             aspect = FIT_HEIGHT;
         } else if   (streq(token, "-fw") || streq(token, "--fit-width")) {
             aspect = FIT_WIDTH;
+        } else if   (streq(token, "-gx") || streq(token, "--gravity-x")) {
+            int last = strlen(args[++i]) - 1;
```

Comment:
the strings in `args` are null-terminated so this check will never pass; you would need to check `last - 1`.

In my opinion, this kind of code checking makes things more complex; if we just enforce users to enter a float between 0 and 1, it's simpler.

Reply:
Just a comment on `last`: `strlen` returns the length of the string _without counting the null terminator_. It means that, when used as an index, it gives you the position of the null terminator. I subtract one, so it's the last character of the string